### PR TITLE
Fix low object LruCaches

### PR DESCRIPTION
**Pull request by benaadams**
## Changes

- Replace was overriting `.Next` and `.Prev` after item had been moved. Carry those values through to new node

#### What types of changes does your code introduce?

- [x] Bugfix (a non-breaking change that fixes an issue)

#### Requires testing

- [x] Yes

#### If yes, did you write tests?

- [x] Yes

### CODE REVIEW

File: src/Nethermind/Nethermind.Core.Test/Caching/LruKeyCacheLowObjectTests.cs [hunk 1]
```diff
@@ -78,6 +78,20 @@ public void Beyond_capacity()
...
+            for (int i = 0; i < Capacity * 2; i++)
+            {
+                for (int ii = 0; ii < Capacity / 2; ii++)
+                {
+                    cache.Set(_addresses[i]);
+                }
+                cache.Set(_addresses[i]);
+            }
+        }
+
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Core.Test/Caching/LruKeyCacheLowObjectTests.cs:94]: no asserts in this tests?

File: src/Nethermind/Nethermind.Core.Test/Caching/LruKeyCacheTests.cs [hunk 2]
```diff
@@ -72,17 +72,31 @@ public void Beyond_capacity()
...
+            for (int i = 0; i < Capacity * 2; i++)
+            {
+                for (int ii = 0; ii < Capacity / 2; ii++)
+                {
+                    cache.Set(_addresses[i]);
+                }
+                cache.Set(_addresses[i]);
+            }
+        }
+
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Core.Test/Caching/LruKeyCacheTests.cs:94]: no asserts in this tests?

**Review by LukaszRozmej — APPROVED** [review]

File: src/Nethermind/Nethermind.Core.Test/Caching/LruKeyCacheLowObjectTests.cs (hunk 1 above)
Comment (benaadams) [reply src/Nethermind/Nethermind.Core.Test/Caching/LruKeyCacheLowObjectTests.cs:94]: throws exception if its wrong

File: src/Nethermind/Nethermind.Core.Test/Caching/LruKeyCacheTests.cs (hunk 2 above)
Comment (benaadams) [reply src/Nethermind/Nethermind.Core.Test/Caching/LruKeyCacheTests.cs:94]: throws exception if its wrong

**Pull request merged**